File: content/snippet/code-signal/arcade/intro/diving-deeper/array-max-consecutive-sum/array-max-consecutive-sum.js
function arrayMaxConsecutiveSum(inputArray, k) {
  let max = 0;
  for (let i = 0; i < inputArray.length - k + 1; i++) {
    let sum = 0;
    for (let j = i; j < i + k; j++) {
      sum += inputArray[j];
    }
    if (sum > max) {
      max = sum;
    }
  }
  return max;
}

console.log(arrayMaxConsecutiveSum([2, 3, 5, 1, 6], 2));

File: content/snippet/code-signal/arcade/intro/diving-deeper/array-max-consecutive-sum/array-max-consecutive-sum.py
def array_max_consecutive_sum(input_list, k):
    max_sum = current_sum = sum(input_list[:k])

    for i in range(len(input_list) - k):
        current_sum += input_list[i + k] - input_list[i]
        max_sum = max(max_sum, current_sum)
    return max_sum


print(array_max_consecutive_sum([2, 3, 5, 1, 6], 2))
